pkg/ipam/multipool.go

```diff
@@ -239,14 +240,32 @@ func newMultiPoolManager(conf Configuration, nodeWatcher nodeWatcher, owner Owne
 	}
 
 	// Subscribe to CiliumNode updates
-	nodeWatcher.RegisterCiliumNodeSubscriber(c)
+	c.eventloop(nodes)
 	owner.UpdateCiliumNodeResource()
 
 	c.waitForAllPools()
 
 	return c
 }
 
+func (m *multiPoolManager) eventloop(res agentK8s.LocalCiliumNodeResource) {
+	// Don't retry events - the downstream code isn't setup to handle retries.
+	// Ensure that the subscrption happens synchronously (but process async), so
+	// that there's no window in which we may miss events.
+	evs := res.Events(context.TODO(), resource.WithErrorHandler(resource.RetryUpTo(0)))
+	go func() {
+		for ev := range evs {
+			switch ev.Kind {
+			case resource.Upsert:
```

Comment:
Yes.

Resource[T] doesn't do any deep equal stuff. The only deduplication that happens is if an event hasn't been observed, and a newer event regarding the same object comes in, you only get one event. This isn't resource code, this is just the k8s workqueue implementation. Since `Resource[T].Events()` fetches the newest state for an object (by key) from the store for each event it delivers anyway, this just means you don't get two identical events.